shell/plugins/global-formatters.js

```diff
@@ -1,10 +1,31 @@
+/* eslint-disable no-console */
 import Vue from 'vue';
 
 const components = require.context('@shell/components/formatter', false, /[A-Z]\w+\.(vue)$/);
 
-components.keys().forEach((fileName) => {
-  const componentConfig = components(fileName);
-  const componentName = fileName.split('/').pop().split('.')[0];
+const globalFormatters = {
+  install: (Vue) => {
+    components.keys().forEach((fileName) => {
+      const componentConfig = components(fileName);
+      const componentName = fileName.split('/').pop().split('.')[0];
 
-  Vue.component(componentName, componentConfig.default || componentConfig);
-});
+      if (Vue.component(componentName)) {
+        // eslint-disable-next-line no-console
+        console.debug(`Skipping ${ componentName } install. Component already exists.`);
+      } else {
+        Vue.component(componentName, componentConfig.default || componentConfig);
+      }
+    });
+  }
+};
+
+export default globalFormatters;
+
+// This is being done for backwards compatibility with our extensions that have written tests and didn't properly make use of Vue.use() when importing and mocking translations
```

Comment:
One more nitpick, these are global formatters and not translations. We could probably modify this comment to be something a little more applicable to more cases. 

```suggestion
// This is being done for backwards compatibility with our extensions that have written tests and didn't properly make use of Vue.use() when importing and mocking vue plugins
```